nemo/collections/tts/modules/audio_codec_modules.py

```diff
@@ -407,40 +417,41 @@ def forward(self, x, l2_norm=False):
         Shapes:
             - x: :math:`(N, 1, T_{in})` or :math:`(N, D_{spec}, T_{in})`
         """
-        x.squeeze_(1)
-        # if you torch spec compute it otherwise use the mel spec computed by the AP
-        if self.use_torch_spec:
-            x = self.torch_spec(x)
+        with default_precision(torch.float32):
```

Comment:
Is this different from using: torch.autocast(device_type=str(audio.device), dtype=torch.float32) ?

Reply:
Yes, it is. We adopted the default precision using the context manager approach, as that’s how PyTorch Lightning handles bf16 precision.

[half.py (lines 54–67)](https://github.com/Lightning-AI/pytorch-lightning/blob/a0134d238a45b3094c07bd0fa717a117858410dd/src/lightning/pytorch/plugins/precision/half.py#L54-L67)

[precision.py (lines 168–190)](https://github.com/Lightning-AI/pytorch-lightning/blob/a0134d238a45b3094c07bd0fa717a117858410dd/src/lightning/pytorch/plugins/precision/precision.py#L168-L190)

I learned this from Piotr.

Reply:
still wondering why should we hack like this. `x` is `bf16`, and `with default_precision` cast it to `fp32` to call `self.torch_spec`. Outside of the block, `x` is cast back to bf16. Should we not to cast `x` to `bf16` before calling `forward`?